strategy-cache-update-and-refresh/controlled.js

```diff
@@ -0,0 +1,46 @@
+var CACHE = 'cache-update-and-refresh';
+
+if ('serviceWorker' in navigator) {
+  navigator.serviceWorker.onmessage = function (evt) {
+    var message = JSON.parse(evt.data);
+
+    var isRefresh = message.type === 'refresh';
+    var isAsset = message.url.includes('asset');
+    var lastETag = localStorage.currentETag;
+    var isNew =  lastETag !== message.eTag;
+
+    if (isRefresh && isAsset && isNew) {
+      // Escape the first time (when there is no ETag yet)
+      if (lastETag) {
+        // Inform the user about the update
+        notice.hidden = false;
+      }
+      localStorage.currentETag = message.eTag;
+    }
+  };
+
+  var notice = document.querySelector('#update-notice');
+
+  var update = document.querySelector('#update');
+  update.onclick = function (evt) {
+    // Avoid navigation.
+    evt.preventDefault();
+    // Open the proper cache.
+    var img = document.querySelector('img');
```

Comment:
can you move this line to the top of the function so it's not in the middle of the comments?